kernel/src/cpu/idt/common.rs

```diff
@@ -314,16 +313,25 @@ struct IdtDesc {
     address: VirtAddr,
 }
 
-#[derive(Copy, Clone, Debug)]
-pub struct IDT {
-    entries: [IdtEntry; IDT_ENTRIES],
+#[derive(Debug)]
+pub struct IDT<'a> {
+    entries: &'a mut [IdtEntry],
 }
 
-impl IDT {
-    pub const fn new() -> Self {
-        IDT {
-            entries: [IdtEntry::no_handler(); IDT_ENTRIES],
-        }
+impl<'a> IDT<'a> {
+    pub const fn new(entries: &'a mut [IdtEntry]) -> Self {
+        Self { entries }
+    }
+
+    /// # Safety
+    /// The caller is required to supply the virtual address of a freshly
+    /// allocated page.
+    pub unsafe fn new_from_page(addr: VirtAddr) -> Self {
+        let idt_ptr = addr.as_mut_ptr::<IdtEntry>();
+        let count = PAGE_SIZE / mem::size_of::<IdtEntry>();
+        // SAFETY; The caller guarantees the suitability of the pointer.
```

Comment:
I think the CI is failing for this line (sorry, it looks like clippy is picky about it):
```suggestion
        // SAFETY: The caller guarantees the suitability of the pointer.
```

Reply:
Thank you for catching my typo!